src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/MicrosoftNetCoreAnalyzersResources.resx

```diff
@@ -2096,4 +2096,28 @@ Widening and user defined conversions are not supported with generic types.</val
   <data name="AvoidSingleUseOfLocalJsonSerializerOptionsTitle" xml:space="preserve">
     <value>Cache and reuse 'JsonSerializerOptions' instances</value>
   </data>
+  <data name="DoNotPassNonNullableValueToArgumentNullExceptionThrowIfNullTitle" xml:space="preserve">
+    <value>Do not pass a non-nullable value to 'ArgumentNullException.ThrowIfNull'</value>
+  </data>
+  <data name="DoNotPassNonNullableValueToArgumentNullExceptionThrowIfNullMessage" xml:space="preserve">
+    <value>Calling 'ArgumentNullException.ThrowIfNull' and passing a non-nullable value does nothing</value>
```

Comment:
NIT: The message could use a bit improvement, though not sure below make it any better
```suggestion
    <value>Calling 'ArgumentNullException.ThrowIfNull' and passing a value that never be null is no-op</value>
```